src/main/scala/sigmastate/serialization/ConstantPlaceholderSerializer.scala

```diff
@@ -0,0 +1,25 @@
+package sigmastate.serialization
+
+import sigmastate.Values._
+import sigmastate._
+import sigmastate.serialization.OpCodes._
+import sigmastate.utils.Extensions._
+import sigmastate.utils.{ByteReader, ByteWriter}
+
+case class ConstantPlaceholderSerializer(cons: (Int, SType) => Value[SType])
+  extends ValueSerializer[ConstantPlaceholder[SType]] {
+
+  override val opCode: OpCode = ConstantPlaceholderIndexCode
+
+  override def serializeBody(obj: ConstantPlaceholder[SType], w: ByteWriter): Unit = {
+    w.putUInt(obj.id)
+    w.putType(obj.tpe)
```

Comment:
@greenhat we probably don't need to save type, as it may be obtained from ConstantStore

Reply:
@ergomorphic Done in [this](https://github.com/ScorexFoundation/sigmastate-interpreter/pull/266/commits/ebccc8b638b5b00e2180c90bddae19845e3864a9) commit.
This makes optional `ByteReader.payload[T]` into required `ByteReader.constantStore`. `ConstantPlaceholderSerializer.parseBody()` will throw if it's missing. 
Also, since ByteReader/Writer API and implementation is due to be extracted into sigma-util does it make sense to move `.constantStore` into a subclass `SigmaByteReader`?

Reply:
@greenhat yes, we can use Sigma-specific interfaces derived from the base R/W interfaces as you suggested.